app/Models/Album.php

```diff
@@ -259,4 +271,58 @@ public function newEloquentBuilder($query): AlbumBuilder
 	{
 		return new AlbumBuilder($query);
 	}
+
+	/**
+	 * Accessor for the "virtual" attribute {@link Album::$track_url}.
+	 *
+	 * This is a convenient method which wraps
+	 * {@link Album::$track_short_path} into
+	 * {@link \Illuminate\Support\Facades\Storage::url()}.
+	 *
+	 * @return string|null the url of the track
+	 */
+	public function getTrackUrlAttribute(): ?string
+	{
+		return $this->track_short_path !== null && $this->track_short_path !== '' ?
+			Storage::url($this->track_short_path) : null;
+	}
+
+	/**
+	 * Set the GPX track for the album.
+	 *
+	 * @param UploadedFile $file the GPX track file to be set
+	 *
+	 * @return void
+	 *
+	 * @throws \Illuminate\Contracts\Filesystem\FileNotFoundException
+	 */
+	public function setTrack(UploadedFile $file): void
+	{
+		if ($this->track_short_path !== null) {
+			Storage::delete($this->track_short_path);
+		}
+
+		$new_track_id = strtr(base64_encode(random_bytes(18)), '+/', '-_');
+		Storage::putFileAs('tracks/', $file, "$new_track_id.xml");
+		$this->track_short_path = "tracks/$new_track_id.xml";
+		$this->save();
+	}
```

Comment:
My  IDE complains about several unhandled/uncaught exceptions here. `random_bytes` may throw `\Exception`, `save()` may throw `ModelDBException`. The latter is fine, because it already inherits the `HttpException` interface, but it should be listed in the PHPdoc. The former is unfortunate, because legacy PHP throws such kind of generic exceptions. I would probably do something like this

```suggestion
	/**
	 * Set the GPX track for the album.
	 *
	 * @param UploadedFile $file the GPX track file to be set
	 *
	 * @return void
	 *
	 * @throws ModelDBException
	 * @throws MediaFileOperationException
	 */
	public function setTrack(UploadedFile $file): void
	{
		try {
			if ($this->track_short_path !== null) {
				Storage::delete($this->track_short_path);
			}

			$new_track_id = strtr(base64_encode(random_bytes(18)), '+/', '-_');
			Storage::putFileAs('tracks/', $file, "$new_track_id.xml");
			$this->track_short_path = "tracks/$new_track_id.xml";
			$this->save();
		} catch (ModelDBException $e) {
			throw $e;
		} catch (\Exception $e) {
			throw new MediaFileOperationException('Could not save track file', $e);
		}
	}
```